src/screens/ConstructionSiteDetailScreen.js

```diff
@@ -39,12 +39,14 @@ const formatDates = (startDate, endDate) => {
   return formattedEndDate ? `${formattedStartDate} - ${formattedEndDate}` : formattedStartDate;
 };
 
+// eslint-disable-next-line complexity
 export const ConstructionSiteDetailScreen = ({ route }) => {
-  const { constructionSites } = useContext(ConstructionSiteContext);
+  const id = route.params?.id;
+  const { constructionSites, loading, refresh, refreshing } = useConstructionSites(id);
 
-  const index = route.params?.index;
+  useMatomoTrackScreenView(`${MATOMO_TRACKING.SCREEN_VIEW.CONSTRUCTION_SITE_DETAIL} / ${id}`);
 
-  if (typeof index !== 'number') return null;
+  if (!id || !constructionSites.length) return null;
```

Comment:
we should not track a screen with an `id` that could be undefined

Reply:
i added a fallback here, which should generally never be used. we only navigate here from the overview screen, and by selecting a specific construction site, which has to have an id.

Reply:
in theory we could, but if we navigate to this screen by accident through other means (static contents or w/e) we might run into problems, because we can not guarantee the id being present.

so in my previous comment i should have phrased it as "we intend to only ever navigate there from the overview screen"

so in our expected cases the `!id` condition will always be false and not influence the outcome of the check. it is still reasonable to keep it there to avoid possible crashes, in the case of the app not being configured properly